loader/loader.go

```diff
@@ -787,3 +788,29 @@ func convertVolumePath(volume types.ServiceVolumeConfig) types.ServiceVolumeConf
 	volume.Source = convertedSource
 	return volume
 }
+
+func resolveEnvironment(dict map[string]any, opts *Options) {
```

Comment:
I was confused reviewing this PR about this func name, which is close to `Project.ResolveEnvironment` but serve a very different scenario, but I don't have a better name for now 😅
Will create a follow-up PR if I find one

Reply:
Ok I'm not alone 😅